src/components/UserSelector.tsx

```diff
@@ -12,24 +60,37 @@ export const UserSelector: React.FC = () => {
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setShowUsers(!showUsers)}
         >
-          <span>Choose a user</span>
+          {targetUser
+            ? (
+              <span>{targetUser.name}</span>
+            ) : (
+              <span>Choose a user</span>
+            )}
```

Comment:
Change only text in tags by condition
`<span>{myCondition ? 'text1' : 'text2'}</span>`
The best way use variable for condition